unstable test in the cpuprofile.TestParallelCPUProfiler

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```

[2022/01/31 17:01:53.595 +00:00] [ERROR] [misc.go:95] ["panic in the recoverable goroutine"] [r="\"send on closed channel\""] ["stack trace"="github.com/pingcap/tidb/util.WithRecovery.func1\n\t/go/tidb/util/misc.go:97\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:1047\nruntime.chansend\n\t/usr/local/go/src/runtime/chan.go:204\nruntime.selectnbsend\n\t/usr/local/go/src/runtime/chan.go:687\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).sendToConsumers\n\t/go/tidb/util/cpuprofile/cpuprofile.go:213\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).doProfiling\n\t/go/tidb/util/cpuprofile/cpuprofile.go:197\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).profilingLoop\n\t/go/tidb/util/cpuprofile/cpuprofile.go:174\ngithub.com/pingcap/tidb/util.WithRecovery\n\t/go/tidb/util/misc.go:100"] [stack="github.com/pingcap/tidb/util.WithRecovery.func1\n\t/go/tidb/util/misc.go:95\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:1047\nruntime.chansend\n\t/usr/local/go/src/runtime/chan.go:204\nruntime.selectnbsend\n\t/usr/local/go/src/runtime/chan.go:687\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).sendToConsumers\n\t/go/tidb/util/cpuprofile/cpuprofile.go:213\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).doProfiling\n\t/go/tidb/util/cpuprofile/cpuprofile.go:197\ngithub.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).profilingLoop\n\t/go/tidb/util/cpuprofile/cpuprofile.go:174\ngithub.com/pingcap/tidb/util.WithRecovery\n\t/go/tidb/util/misc.go:100"]

panic: test timed out after 20m0s
69303
69304
goroutine 51 [running]:
69305
testing.(*M).startAlarm.func1()
69306
	/usr/local/go/src/testing/testing.go:1788 +0xbb
69307
created by time.goFunc
69308
	/usr/local/go/src/time/sleep.go:180 +0x4a
69309
69310
goroutine 1 [chan receive, 20 minutes]:
69311
testing.(*T).Run(0xc00047bba0, {0x15cc0be, 0x17}, 0x1621c50)
69312
	/usr/local/go/src/testing/testing.go:1307 +0x752
69313
testing.runTests.func1(0x0)
69314
	/usr/local/go/src/testing/testing.go:1598 +0x9a
69315
testing.tRunner(0xc00047bba0, 0xc000409ae8)
69316
	/usr/local/go/src/testing/testing.go:1259 +0x230
69317
testing.runTests(0xc0006fea00, {0x22ca120, 0x4, 0x4}, {0xc000409bd0, 0x4406dd, 0x22f2e80})
69318
	/usr/local/go/src/testing/testing.go:1596 +0x7cb
69319
testing.(*M).Run(0xc0006fea00)
69320
	/usr/local/go/src/testing/testing.go:1504 +0x9d2
69321
go.uber.org/goleak.VerifyTestMain({0x19a1c20, 0xc0006fea00}, {0x0, 0x0, 0x0})
69322
	/home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x5a
69323
github.com/pingcap/tidb/util/cpuprofile.TestMain(0x400)
69324
	/go/tidb/util/cpuprofile/cpuprofile_test.go:40 +0x59
69325
main.main()
69326
	_testmain.go:51 +0x265
69327
69328
goroutine 8 [semacquire, 20 minutes]:
69329
sync.runtime_SemacquireMutex(0xc0004aa060, 0x0, 0x0)
69330
	/usr/local/go/src/runtime/sema.go:71 +0x25
69331
sync.(*Mutex).lockSlow(0xc0004aa060)
69332
	/usr/local/go/src/sync/mutex.go:138 +0x21d
69333
sync.(*Mutex).Lock(0xc0004aa060)
69334
	/usr/local/go/src/sync/mutex.go:81 +0x65
69335
github.com/pingcap/tidb/util/cpuprofile.(*parallelCPUProfiler).consumersCount(0xc0004aa060)
69336
	/go/tidb/util/cpuprofile/cpuprofile.go:203 +0x33
69337
github.com/pingcap/tidb/util/cpuprofile.TestParallelCPUProfiler(0x0)
69338
	/go/tidb/util/cpuprofile/cpuprofile_test.go:106 +0x512
69339
testing.tRunner(0xc0000ba680, 0x1621c50)
69340
	/usr/local/go/src/testing/testing.go:1259 +0x230
69341
created by testing.(*T).Run
69342
	/usr/local/go/src/testing/testing.go:1306 +0x727

```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/cpuprofile/cpuprofile.go
+++ b/util/cpuprofile/cpuprofile.go
@@ -25,6 +25,7 @@ import (
 	"github.com/pingcap/tidb/metrics"
 	"github.com/pingcap/tidb/util"
 	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
 )
 
 // DefProfileDuration exports for testing.
@@ -57,6 +58,7 @@ func StopCPUProfiler() {
 // Normally, the registered ProfileConsumer will receive the cpu profile data per second.
 // If the ProfileConsumer (channel) is full, the latest cpu profile data will not be sent to it.
 // This function is thread-safe.
+// WARN: ProfileConsumer should not be closed before unregister.
 func Register(ch ProfileConsumer) {
 	globalCPUProfiler.register(ch)
 }
@@ -208,6 +210,13 @@ func (p *parallelCPUProfiler) consumersCount() int {
 
 func (p *parallelCPUProfiler) sendToConsumers() {
 	p.Lock()
+	defer func() {
+		p.Unlock()
+		if r := recover(); r != nil {
+			logutil.BgLogger().Error("parallel cpu profiler panic", zap.Any("recover", r))
+		}
+	}()
+
 	for c := range p.cs {
 		select {
 		case c <- p.profileData:
@@ -216,5 +225,4 @@ func (p *parallelCPUProfiler) sendToConsumers() {
 		}
 	}
 	p.profileData = nil
-	p.Unlock()
 }

--- a/util/cpuprofile/cpuprofile_test.go
+++ b/util/cpuprofile/cpuprofile_test.go
@@ -177,13 +177,13 @@ func TestGetCPUProfile(t *testing.T) {
 	defer cancel()
 	testutil.MockCPULoad(ctx, "sql", "sql_digest", "plan_digest")
 	var wg sync.WaitGroup
-	for i := 0; i < 3; i++ {
+	for i := 0; i < 10; i++ {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
 			var err error
 			buf := bytes.NewBuffer(nil)
-			err = getCPUProfile(time.Millisecond*400, buf)
+			err = getCPUProfile(time.Millisecond*1000, buf)
 			require.NoError(t, err)
 			profileData, err := profile.Parse(buf)
 			require.NoError(t, err)

